packages/component_code_gen/code_gen/generate_component_code.py

```diff
@@ -4,26 +4,44 @@
 import helpers.langchain_helpers as langchain_helpers
 import helpers.supabase_helpers as supabase_helpers
 
+from config.config import config
 import config.logging_config as logging_config
 logger = logging_config.getLogger(__name__)
 
 
-def main(app, prompt, templates):
+def generate_code(app, prompt, templates):
+    validate_inputs(app, prompt, templates)
+
     db = supabase_helpers.SupabaseConnector()
 
-    docs_meta = db.get_app_docs_meta(app)
+    auth_meta = db.get_app_auth_meta(app)
+    add_code_example(templates, auth_meta['component_code_scaffold_raw']) # TODO: is this needed only for actions?
```

Comment:
I had trouble tracing where this code actually added the example code in the template

Reply:
It's right on the [end of the template](https://github.com/PipedreamHQ/pipedream/pull/7714/files/e60deeb82603512f2d4c2a4fd143d89821277c14#diff-7017b494ad167ef6f8c0143e27f3df3e38bd4fbff1d4509db5b67fbdeab6c7cbR425). I might end up moving this to somewhere higher up